test/Have.lean

```diff
@@ -43,3 +44,10 @@ example : True := by
   have _q
   · exact 6
   simp
+
+/--
+error: type expected, got
+  (Eq : ?m.452 → ?m.452 → Prop)
+-/
+#guard_msgs in
+example : True := by have h : Eq
```

Comment:
To make the test be more tolerant to metavariable numbering changes, should we do `have h : Nat.zero` or something?

I'm thinking of adding a `renumber_mvars` option to `#guard_msgs` so that we don't have to worry about this problem in the future. It'd just textually look for `?m.` followed by some number of digits, and then renumber them starting with 1, across all the messages rather than one-by-one. It's a hack, but maybe it'll do.

Reply:
I'm aware of the work being done on this, but it seems orthogonal to this PR. If the situation improves we can use it here.

Reply:
(I didn't mean to suggest that this PR needed to wait for any changes to `#guard_msgs` -- it was just a passing design thought while you were here.)